compiler/rustc_hir_typeck/src/fn_ctxt/suggestions.rs

```diff
@@ -3533,6 +3533,12 @@ impl<'a, 'tcx> FnCtxt<'a, 'tcx> {
                 .must_apply_modulo_regions()
         {
             let sm = self.tcx.sess.source_map();
+            // If the span of rhs_expr is in an external macro,
+            // we should not suggest to swap the equality. See issue #139050
+            if rhs_expr.span.in_external_macro(sm) {
```

Comment:
This needs to check both `lhs_expr` and `rhs_expr`, since suggestions are invalid if either are in an external macro.